spec/lib/importers/licence_transaction/licence_importer_spec.rb

```diff
@@ -0,0 +1,189 @@
+    )
+  end
+
+  context "when a licence is valid" do
+    before do
+      stub_publishing_api_has_links(
+        { content_id: "46044b4d-a41b-42c1-882d-8d03e65f24cd", links: publising_api_get_links_response },
+      )
+    end
+
+    it "migrates the licence" do
+      put_content_request = stub_publishing_api_put_content(
+        new_content_id, expected_put_content_payload
+      )
+      publish_request = stub_publishing_api_publish(
+        new_content_id, { update_type: "republish", locale: "en" }
+      )
+      patch_links_request = stub_publishing_api_patch_links(
+        new_content_id, { links: expected_patch_links_payload }
+      )
+
+      expect { described_class.new.call }
+        .to output(successful_import_message).to_stdout
+
+      expect(put_content_request).to have_been_requested
+      expect(patch_links_request).to have_been_requested
+      expect(publish_request).to have_been_requested
+    end
+  end
+
+  context "when a licence is invalid" do
+    let(:publishing_api_response) do
+      publishing_api_licences_response.tap { |licences| licences.first["title"] = nil }
+    end
+
+    it "doesn't migrate the licence" do
+      expect { described_class.new.call }
+        .to output(invalid_licence_error_message).to_stdout
+
+      expect(stub_any_publishing_api_put_content).to_not have_been_requested
+      expect(stub_any_publishing_api_patch_links).to_not have_been_requested
+      expect(stub_any_publishing_api_publish).to_not have_been_requested
+    end
+  end
+
+  context "when a licence isn't present in the common licences list" do
+    let(:publishing_api_response) do
+      publishing_api_licences_response.tap do |licences|
+        licences.first["details"]["licence_identifier"] = "1111-2-3"
+      end
+    end
+
+    it "doesn't migrate the licence" do
+      described_class.new.call
+
+      expect(stub_any_publishing_api_put_content).to_not have_been_requested
+      expect(stub_any_publishing_api_patch_links).to_not have_been_requested
+      expect(stub_any_publishing_api_publish).to_not have_been_requested
+    end
+  end
+
+  def invalid_licence_error_message
+    "[ERROR] licence: /find-licences/art-therapist-registration has validation errors: #<ActiveModel::Errors [#<ActiveModel::Error attribute=title, type=blank, options={}>]>\n"
+  end
+
+  def successful_import_message
+    "Published: /find-licences/art-therapist-registration\n"
+  end
+
+  def expected_put_content_payload
+    {
+      base_path: "/find-licences/art-therapist-registration",
+      title: "Art therapist registration",
+      description: "You need to register with the Health and Care Professions Council (HCPC) to practise as an art therapist in the UK",
+      document_type: "licence_transaction",
+      change_note: "Imported from Publisher",
+      schema_name: "specialist_document",
+      publishing_app: "specialist-publisher",
+      rendering_app: "government-frontend",
```

Comment:
Should this now be "frontend" now that https://github.com/alphagov/specialist-publisher/pull/2184 has been merged?